instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -159,19 +163,37 @@ def _traced_execute_command(func, instance, args, kwargs):
             return response
 
     def _traced_execute_pipeline(func, instance, args, kwargs):
-        cmds = [_format_command_args(c) for c, _ in instance.command_stack]
+        try:
+            command_stack = (
+                instance.command_stack
+                if hasattr(instance, "command_stack")
+                else instance._command_stack
+            )
+        except AttributeError:
+            command_stack = []
+
+        cmds = [
+            _format_command_args(c.args if hasattr(c, "args") else c[0])
+            for c in command_stack
+        ]
         resource = "\n".join(cmds)
 
-        span_name = " ".join([args[0] for args, _ in instance.command_stack])
+        span_name = " ".join(
+            [
+                (c.args[0] if hasattr(c, "args") else c[0][0])
+                for c in command_stack
+            ]
+        )
```

Comment:
I would also suggest to safe guard this code for potential `IndexError`s.

Reply:
updated to handle `IndexError`.

Reply:
updated to handle `AttributeError` and `IndexError` in one try block.